lib/client.js

```diff
@@ -29,12 +29,22 @@ function defaultId () {
 }
 
 function sendPacket (client, packet, cb) {
+  function emitAndCb () {
+    if ('publish' === packet.cmd) {
+      client.emit('publish', packet.topic, packet.payload, packet);
+    }
+    if (cb) {
+      cb();
+    }
+  }
```

Comment:
Given how the logic work, we can generate a wrapping function only if `write` returns false.

However, for logging purposes, we might even emit it anyway even if `write` returns false.